internal/digraph/scheduler/scheduler.go

```
@@ -284,7 +287,12 @@ func (sc *Scheduler) Schedule(ctx context.Context, graph *ExecutionGraph, progre
 								node.SetStatus(NodeStatusSuccess)
 							} else {
 								node.MarkError(execErr)
-								sc.setLastError(execErr)
+								// Only set lastError if this failure should not be allowed to continue
+								// If shouldContinue() is true, this error is allowed and should contribute
+								// to partial success rather than overall failure
+								if !node.shouldContinue(ctx) {
+									sc.setLastError(execErr)
+								}
```

Not entirely sure about this one. @yottahmd - should last error be recorded for ignored errors? I had to change it to fix the scheduler tests.

---

This looks good to me! Thanks for your careful attention to detail.